frontend/public/components/monitoring/alert-manager-receiver-forms.tsx

```diff
@@ -200,143 +210,136 @@ const RoutingLabels = ({ routeLabels, setRouteLabels, saveAsDefaultReceiver }) =
+  let initialWebhookUrl = '';
+
+  let receiverToEdit: AlertManagerReceiver;
+  if (editReceiverNamed) {
+    receiverToEdit = _.find(_.get(config, 'receivers'), { name: editReceiverNamed });
+    initialReceiverName = receiverToEdit ? receiverToEdit.name : '';
+    initialReceiverType = _.find(_.keys(receiverToEdit), (key) => _.includes(key, '_configs'));
+    if (initialReceiverType) {
+      let receiverConfig;
+      initialReceiverType = initialReceiverType.slice(0, initialReceiverType.indexOf('_'));
+      switch (initialReceiverType) {
+        case 'pagerduty':
+          receiverConfig = _.get(receiverToEdit, 'pagerduty_configs[0]');
+          initialPagerDutyValues = {
+            integrationType: _.get(receiverConfig, 'routing_key') ? 'events' : 'prometheus',
+            integrationKey:
+              _.get(receiverConfig, 'routing_key') || _.get(receiverConfig, 'service_key'),
+          };
+          break;
+        case 'webhook':
+          receiverConfig = _.get(receiverToEdit, 'webhook_configs[0]');
+          initialWebhookUrl = _.get(receiverConfig, 'url');
+          break;
+        default:
+      }
+    }
+  }
+
+  const [receiverName, setReceiverName] = React.useState(initialReceiverName);
+  const [receiverType, setReceiverType] = React.useState(initialReceiverType);
+  const [webhookUrl, setWebhookUrl] = React.useState(initialWebhookUrl);
+  const [pagerDutyValues, setPagerDutyValues] = React.useState(initialPagerDutyValues);
 
   const { route } = config || {};
-  const { receiver: defaultReceiver } = route || {};
-  const saveAsDefaultReceiver = _.isUndefined(defaultReceiver);
-  const defaultRouteLabels = saveAsDefaultReceiver
-    ? [{ name: 'All (default receiver)', value: 'All (default receiver)', isRegex: false }]
-    : [{ name: '', value: '', isRegex: false }];
-  const [routeLabels, setRouteLabels] = React.useState(defaultRouteLabels);
-
-  /**
-   * Converts array of labels:
-   * [
-   *   {
-   *     "name": "severity",
-   *     "value": "warning",
-   *     "isRegex": false
-   *   },
-   *   {
-   *    "name": "cluster",
-   *     "value": "myCluster",
-   *     "isRegex": false
-   *   }
-   * ]
-   * ...to single label object:
-   * {
-   *   "severity": "warning",
-   *   "cluster": "myCluster"
-   * }
-   */
-  const normalizeRouteLabels = (returnRegexMatches = false) => {
-    const matches = _.filter(routeLabels, (routeLabel) =>
-      returnRegexMatches ? routeLabel.isRegex : !routeLabel.isRegex,
-    );
-    const matchObj = _.map(matches, (label) => {
-      return { [label.name]: label.value };
-    });
-    return _.assign({}, ...matchObj);
-  };
+  const { receiver: defaultReceiver } = route || {}; // top level route.receiver is the default receiver for all alarms
+  // if no default receiver defined or editing the default receiver
+  const isDefaultReceiver = _.isUndefined(defaultReceiver) || defaultReceiver === editReceiverNamed;
```

Comment:
Nit: Probably better to use `=== undefined` instead of `_.isUndefined`. I know we use Lodash everywhere, but better to avoid it when there's no real benefit.